server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonRecordReader.java

```diff
@@ -241,21 +245,38 @@ private boolean scanToNextJsonBeginObject() throws IOException {
         // seek until we hit the first begin-object
         boolean inString = false;
         int i;
+        StringBuilder readValues = new StringBuilder();
         // since we have not yet found a starting object, exit if either EOF (-1) or END_OF_SPLIT (-2)
         while ((i = readNextChar()) > EOF) {
             char c = (char) i;
+
+            // while scanning to start of object, we need to make sure we aren't losing our position
+            readValues.append(c);
+            checkLength(readValues, 1024);
```

Comment:
these 2 lines repeat with a magic number, can we extract into a function, take only 1 param and declare a const for a number ?
```suggestion
            accountForCharacterRead(c);
```